app/migrate.go

```diff
@@ -0,0 +1,237 @@
+
+			newGenState[ibcxfertypes.ModuleName] = clientCtx.JSONMarshaler.MustMarshalJSON(ibcTransferGenesis)
+			newGenState[host.ModuleName] = clientCtx.JSONMarshaler.MustMarshalJSON(ibcCoreGenesis)
+			newGenState[captypes.ModuleName] = clientCtx.JSONMarshaler.MustMarshalJSON(capGenesis)
+			newGenState[evtypes.ModuleName] = clientCtx.JSONMarshaler.MustMarshalJSON(evGenesis)
+			newGenState[staking.ModuleName] = clientCtx.JSONMarshaler.MustMarshalJSON(&stakingGenesis)
+
+			genDoc.AppState, err = json.Marshal(newGenState)
+			if err != nil {
+				return errors.Wrap(err, "failed to JSON marshal migrated genesis state")
+			}
+
+			genesisTime, _ := cmd.Flags().GetString(flagGenesisTime)
+			if genesisTime != "" {
+				var t time.Time
+
+				err := t.UnmarshalText([]byte(genesisTime))
+				if err != nil {
+					return errors.Wrap(err, "failed to unmarshal genesis time")
+				}
+
+				genDoc.GenesisTime = t
+			}
+
+			chainID, _ := cmd.Flags().GetString(flags.FlagChainID)
+			if chainID != "" {
+				genDoc.ChainID = chainID
+			}
+
+			initialHeight, _ := cmd.Flags().GetInt(flagInitialHeight)
+
+			genDoc.InitialHeight = int64(initialHeight)
+
+			replacementKeys, _ := cmd.Flags().GetString(flagReplacementKeys)
+
+			if replacementKeys != "" {
+				genDoc = loadKeydataFromFile(clientCtx, replacementKeys, genDoc)
+			}
+
+			bz, err := tmjson.Marshal(genDoc)
+			if err != nil {
+				return errors.Wrap(err, "failed to marshal genesis doc")
+			}
+
+			sortedBz, err := sdk.SortJSON(bz)
+			if err != nil {
+				return errors.Wrap(err, "failed to sort JSON genesis doc")
+			}
+
+			fmt.Println(string(sortedBz))
+			return nil
+		},
+	}
+
+	cmd.Flags().String(flagGenesisTime, "", "override genesis_time with this flag")
+	cmd.Flags().Int(flagInitialHeight, 0, "Set the starting height for the chain")
+	cmd.Flags().String(flagReplacementKeys, "", "Proviide a JSON file to replace the consensus keys of validators")
+	cmd.Flags().String(flags.FlagChainID, "", "override chain_id with this flag")
+	cmd.Flags().Bool(flagNoProp29, false, "Do not implement fund recovery from prop29")
+
+	return cmd
+}
+
+// MigrateTendermintGenesis makes sure a later version of Tendermint can parse
+// a JSON blob exported by an older version of Tendermint.
+func migrateTendermintGenesis(jsonBlob []byte) ([]byte, error) {
+	var jsonObj map[string]interface{}
+	err := json.Unmarshal(jsonBlob, &jsonObj)
+	if err != nil {
+		return nil, err
+	}
+
+	consensusParams, ok := jsonObj["consensus_params"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("exported json does not contain consensus_params field")
+	}
+	evidenceParams, ok := consensusParams["evidence"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("exported json does not contain consensus_params.evidence field")
+
+	}
+
+	evidenceParams["max_age_num_blocks"] = evidenceParams["max_age"]
+	delete(evidenceParams, "max_age")
+
+	evidenceParams["max_age_duration"] = "172800000000000"
+
```

Comment:
```suggestion
	evidenceParams["max_age_duration"] = "172800000000000"
	evidenceParams["max_bytes"] = "100000"

```
I think this parameter needs to be set, otherwise evidence won't be allowed to be submitted (I think it is a newly introduced param like max age duration). I set it to `100000` which is half of the current block size `max_bytes` (200000). Maybe it would be better to parse the block consensus param max_bytes and set the evidence to be a certain percentage of that?

Reply:
quick calculations from @cmwaters 
```
The max size light client evidence can be is 1,111,485 bytes (with 10,000 validators). For 150 validators it is 18,135 so maybe 50,000 bytes is reasonable
```